packages/schematics/angular/utility/dependencies.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { JsonAstObject, JsonParseMode, parseJsonAst } from '@angular-devkit/core';
+import { SchematicsException, Tree } from '@angular-devkit/schematics';
+import {
+  appendPropertyInAstObject,
+  findPropertyInAstObject,
+} from './json-utils';
+
+
+const pkgJsonPath = '/package.json';
+export enum NodeDependencyType {
+  Default,
+  Dev,
+  Peer,
+  Optional,
+}
+
+export interface NodeDependency {
+  type: NodeDependencyType;
+  packageName: string;
```

Comment:
just `name` would probably be better since it can be a package name, file, GitHub repo, etc.